File: frontend/services/feed.ts
/** Live activity feed API service (spec §8). */

import { apiGet } from "@/lib/api";

export interface FeedItem {
  id: string;
  kind: "change_event" | "score_increase";
  brewery_id: string;
  brewery_name: string;
  brewery_slug: string;
  summary: string;
  details: Record<string, unknown>;
  created_at: string;
}

export interface FeedPage {
  items: FeedItem[];
  total: number;
  limit: number;
  offset: number;
}

export function fetchFeed(params?: {
  limit?: number;
  offset?: number;
}): Promise<FeedPage> {
  return apiGet<FeedPage>("/feed", params);
}

File: backend/app/schemas/feed.py
"""Live activity feed schemas (spec §8)."""

from __future__ import annotations

import uuid
from datetime import datetime
from typing import Any, Literal

from pydantic import BaseModel

FeedItemKind = Literal["change_event", "score_increase"]


class FeedItem(BaseModel):
    """A single cross-brewery activity item."""

    id: uuid.UUID
    kind: FeedItemKind
    brewery_id: uuid.UUID
    brewery_name: str
    brewery_slug: str
    summary: str
    details: dict[str, Any]
    created_at: datetime


class FeedPage(BaseModel):
    """A paginated slice of the activity feed."""

    items: list[FeedItem]
    total: int
    limit: int
    offset: int
